src/main/java/com/scottlogic/AuthorPostSorter.java

```diff
@@ -0,0 +1,21 @@
+package com.scottlogic;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+public class AuthorPostSorter implements PostSorter {
+
+    public List<UserPost> sort(List<UserPost> inputList) {
+        if (inputList == null) {
+            return Arrays.asList();
+        }
+
+        if (inputList.size() == 0) {
+            return inputList;
+        }
```

Comment:
You can simplify this to something like
```
if (inputList == null || inputList.emptyList()) {
  return Collections.emptyList();
}
```

Reply:
Originally I did have this but not sure why I did it like that after - fixed